helper_crates/vtable/src/lib.rs

```diff
@@ -303,6 +303,11 @@ impl<'a, T: ?Sized + VTableMeta> VRef<'a, T> {
             None
         }
     }
+
+    /// Returns a pointer to the VRef's instance. This is primarily useful for comparisons.
+    pub fn as_ptr(this: Self) -> NonNull<u8> {
```

Comment:
I'm curious, why does this consume Self instead of being `&self` and then use `self.inner.ptr.clone()`?

I'm asking because I see other `as_ptr` functions in the standard library not consume.

Reply:
VRef is Copy, so that's not really consuming

I'm not sure what to do here to be honest.

NonNull::as_ptr  consume:
https://doc.rust-lang.org/stable/std/ptr/struct.NonNull.html#method.as_ptr

pointer::as_ptr too:
https://doc.rust-lang.org/stable/std/primitive.pointer.html#method.as_ptr

What do you think?

Reply:
Ah, I missed the Copy part. Then the "consuming" signature makes sense and it's consistent.